tbot_contrib/time_generic_testcases/time_tescases.py

```diff
@@ -0,0 +1,101 @@
+from termcolor import colored
+import statistics
+import time
+import tbot
+import typing
+
+
+@tbot.testcase
+def testcase_clockTime(
+    testcase: typing.Callable, *args: typing.Any, **kwargs: typing.Any
+) -> float:
+    """Return the time that the given testcase took
+    you can pass any arguments that the testcase needs"""
+    start = time.time()
+    testcase(*args, **kwargs)
+    finish = time.time()
+    return finish - start
+
+
+@tbot.testcase
+def statistics_clockTime(
+    testcase: typing.Callable, *args: typing.Any, **kwargs: typing.Any
+) -> None:
+    """receive a testcase and prints descriptive statics
+    you can pass as an argument:
+    times = the number of times that the testcase is to be repeated
+    sleep = the time in seconds that the bucle must wait
+
+    you can pass any arguments that the testcase needs """
+
+    if "times" in kwargs:
+        times = kwargs.pop("times")
+    else:
+        times = 100
+
+    if "sleep" in kwargs:
+        sleep = kwargs.pop("sleep")
+    else:
+        sleep = 0
+
+    arrayElapsedTimes = []
+
+    for n in range(times):
+
+        elapesedTime = testcase_clockTime(testcase, *args, **kwargs)
+        arrayElapsedTimes.append(elapesedTime)
+        time.sleep(sleep)
+
+    mean = statistics.mean(arrayElapsedTimes)
+    harmonicMean = statistics.harmonic_mean(arrayElapsedTimes)
+    median = statistics.median(arrayElapsedTimes)
+    variance = statistics.pvariance(arrayElapsedTimes)
+    standardDeviation = statistics.pstdev(arrayElapsedTimes)
```

Comment:
Could you add a [named tuple](https://docs.python.org/3/library/collections.html#collections.namedtuple) for these values and fill it here instead of creating a lot of local variables?  You could then also return the named tuple so a calling testcase could make use of the values.  E.g.:

```python
# At the top of the file
import collections


TimingResults = collections.namedtuple("TimingResults", ["mean", "harmonic_mean", "median", "pvariance", "pstdev"])

# And in the testcase
results = TimingResults(
    statistics.mean(elapsed_times),
    statistics.harmonic_mean(elapsed_times),
    statistics.median(elapsed_times),
    statistics.pvariance(elapsed_times),
    statistics.pstdev(elapsed_times),
)
```